vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.ts

```diff
@@ -477,11 +471,6 @@ window.Vaadin.Flow.gridConnector.initLazy = (grid) => {
         updateGridCache(page);
       }
     }
-    let cacheToClear = dataProviderController.rootCache;
-    const endIndex = index + updatedPageCount * grid.pageSize;
-    for (let itemIndex = index; itemIndex < endIndex; itemIndex++) {
-      delete cacheToClear.items[itemIndex];
```

Comment:
This additional clearing was necessary because `updateGridCache` relied on `rootCache.setPage`, which only updated indexes smaller than the current size. In other words, it failed to remove exceeding items if the size was decreased before calling `$connector.clear`.